lib/poetry.js

```diff
@@ -8,24 +8,25 @@ const tomlParse = require('@iarna/toml/parse-string');
 /**
  * poetry install
  */
-async function pyprojectTomlToRequirements() {
-  if (!this.options.usePoetry || !isPoetryProject(this.servicePath)) {
+async function pyprojectTomlToRequirements(modulePath, pluginInstance) {
+  const { serverless, servicePath, options, log, progress } = pluginInstance;
+
+  const moduleProjectPath = path.join(servicePath, modulePath);
+  if (!options.usePoetry || !isPoetryProject(moduleProjectPath)) {
     return;
   }
 
   let generateRequirementsProgress;
-  if (this.progress && this.log) {
-    generateRequirementsProgress = this.progress.get(
+  if (progress && log) {
+    generateRequirementsProgress = progress.get(
```

Comment:
Did you validate how that behaves when there's multiple functions packaged individually at the same time? Are they being packaged in parallel or not?

Reply:
No, I was not aware of this mechanism.  I will try to investigate and get back to you.

Reply:
I conducted an experiment with multiple functions packaged individually.  It appears that this plugin is written to package the functions sequentially and does not support parallel packaging:

From `installAllRequirements.js`:
```
    for (const f of filteredFuncs) {
      if (!get(f, 'module')) {
        set(f, ['module'], '.');
      }

      // If we didn't already process a module (functions can re-use modules)
      if (!doneModules.includes(f.module)) {
        const reqsInstalledAt = await installRequirementsIfNeeded(
          f.module,
          f,
          this
        );
...
```

Note how the `await installRequirementsIfNeeded` is inside of the `for` loop.  This will prevent parallel execution of the packaging of the functions.

Reply:
Thanks for checking that 🙇